internal/v2/tls_config_store/tls_config_store_test.go

```diff
@@ -0,0 +1,151 @@
+package tls_config_store
+
+import (
+	"testing"
+	"crypto/tls"
+	"bytes"
+
+	_ "embed"
+)
+
+var (
+	//go:embed example_cert_key/client_cert.pem
+	clientCertpem []byte
+	//go:embed example_cert_key/server_cert.pem
+	serverCertpem []byte
+	//go:embed example_cert_key/client_key.pem
+	clientKeypem []byte
+	//go:embed example_cert_key/server_key.pem
+	serverKeypem []byte
+)
+
+
+// TODO(riyamehta) : In Client and Server test, verify contents of config.RootCAs once x509.CertPool.Equal function is officially released : https://cs.opensource.google/go/go/+/4aacb7ff0f103d95a724a91736823f44aa599634 .
+
+func TestTLSConfigStoreClient(t *testing.T) {
+	// Setup for static client test.
+	cert, err := tls.X509KeyPair(clientCertpem, clientKeypem)
+	if err != nil {
+		t.Errorf("Test suite setup failed")
+	}
+	// certPool := x509.NewCertPool()
+        // certPool.AppendCertsFromPEM(serverCertpem)
+
+	for _, tc := range []struct {
+		description		    string
+		Certificates                []tls.Certificate
+		// RootCAs 		    *x509.CertPool
+		InsecureSkipVerify          bool
+		CipherSuites	            []uint16
+		ClientSessionCache	    tls.ClientSessionCache
+		MinVersion	            uint16
+		MaxVersion		    uint16
+		CurvePreferences	    []tls.CurveID
+	}{
+		{
+			description: "static",
```

Comment:
Would it be worthwhile to a nil test(Certificates == nil) to test this static implementation?

Reply:
Sorry I missed this - I think it's OK to omit this for now, since we'll be deferring this to S2A in the future.

Reply:
Sounds good, thanks!